utils/eventtracking.js

```diff
@@ -0,0 +1,83 @@
+const MIXPANEL_TOKEN = '9aa8926fbcb03eb5d6ce787b5e8fa6eb';
+var mixpanel = require('mixpanel').init(MIXPANEL_TOKEN);
+
+const uuid = require('uuid');
+const chalk = require('chalk');  // colorize output
+const readline = require('readline');
+const settings = require('./settings');
+
+const TRACKING_ENABLED_KEY = 'trackingEnaled';
+const TRACKING_SESSION_ID_KEY = 'trackingSessionId';
+
+const track = async (eventType, eventProperties) => {
+    const shellSettings = settings.getShellSettings();
+    // if the appropriate option is not in settings, ask now and save settings.
+    if (!(TRACKING_ENABLED_KEY in shellSettings)) {
+        const rl = readline.createInterface({
+            input: process.stdin,
+            output: process.stdout
+        });
+        const getEventTrackingConsent = async () => {
+            for (var attempts = 0; attempts < 10; attempts++) {
+                const answer = await new Promise((resolve) => {
+               
+                    rl.question(
+                        chalk`We would like to collect data on near shell usage to improve developer experience. {bold.yellow Would you like to opt in (y/n)?}`,
```

Comment:
we should probably talk more information about what we are collecting and how we are planning to use this.

Reply:
@potatodepaulo could we get some final wording for this?

Reply:
How about:
> We would like to collect data on near-shell usage to improve developer experience. We will never send private information. We only collect which commands are run via an anonymous identifier.

Reply:
The more text you put in here, the less people are going to read. For now let’s use Mike’s suggestion, and then we can add a link to a data privacy page with a full explanation later.